cutgeneratingfunctionology/spam/basic_semialgebraic.py

```diff
@@ -1015,13 +1017,17 @@ def eq_poly(self):
 
         Together, ``eq_poly``, ``lt_poly``, and ``le_poly`` describe ``self``.
         """
-        for c in self._polyhedron.minimized_constraints():
-            if c.is_equality():
-                coeff = c.coefficients()
-                # observe: coeffients in a constraint of NNC_Polyhedron could have gcd != 1. # take care of this.
-                gcd_c = gcd(gcd(coeff), c.inhomogeneous_term())
-                t = sum(QQ(x)/gcd_c*y for x, y in zip(coeff, self.poly_ring().gens())) + QQ(c.inhomogeneous_term())/gcd_c
-                yield t
+        ## add tests
```

Comment:
Why was this change (test for is_empty) necessary again? Is something wrong with the minimized_constraints?

Reply:
I don't remember exactly. I think it was there as a fix from before we knew that BasicSemialgebraicSet_section‎ wasn't stable and when we had fixed that. I've done a few quick tests and it seems like the test for emptiness is not longer necessary.

Reply:
Let's back out this change then